build/builder.go

```diff
@@ -117,6 +117,9 @@ type Options struct {
 	//
 	// Note: heap checking is "best effort", and it's possible for the process to OOM without triggering the heap profile.
 	HeapProfileTriggerBytes uint64
+
+	// UseSourcegraphIDForName is true if we want to use the Sourcegraph ID as prefix for the shard name.
+	UseSourcegraphIDForName bool
```

Comment:
maybe instead we should have an optional string here which controls the prefix for shardNameVersion? indexserver then passes in the name via a string? Then its less sg specific/more flexible. WDYT?

Reply:
Good idea.